src/v/cloud_storage/remote.cc

```diff
@@ -537,4 +553,73 @@ ss::future<download_result> remote::segment_exists(
     co_return *result;
 }
 
+ss::future<>
+remote::propagate_credentials(cloud_roles::credentials credentials) {
+    return container().invoke_on_all([c = credentials](remote& svc) mutable {
```

Comment:
I thought that the credentials will be fetched only on shard 0 but it looks like `maybe_start_auth_refresh_op` is invoked on every shard.

Reply:
Yes it is correct that credentials are fetched on a fixed shard, my plan was that `maybe_start_auth_refresh_op` is run on all shards, then:

1. if we are using static credentials, the callback is triggered once to supply the credentials to the s3 client pool of the shard 
2. if we are using dynamic credentials we check for the shard id and then trigger the background operation in `do_start_auth_refresh_op`.

 I think there is a bug I introduced here. the code should use `pool.local().load...` via remote, instead of the callback in case of static credentials, otherwise each shard will just copy the credentials to each other shard. 

Thanks for pointing out, I will fix it.

Reply:
I changed it so that we only do operations on shard 0, all other shards will just receive credentials from shard 0, in both static and dynamic credentials case.